Changed ImageProcess to a Builder.

nspitzern opened the pull request:
ImageProcess now contains a state of the image and does the operations on it.
We can concatenate operations and do them one we one call.

nspitzern committed:
Changed ImageProcess to a Builder.
ImageProcess now contains a state of the image and does the operations on it.
We can concatenate operations and do them one we one call.

shimoncohen reviewed (changes requested):
In the following code:
https://github.com/nspitzern/Image-Process-Project/blob/69a4fa27c4815476659e5632348ed30d180cc41f/src/ImageProcess/ImageProcess.java#L178
Why does FFT inverse receive FFT?
The reverse is supposed to be on this.image.

Suggestion:
```java
public Image inverseFFT() {
    FFT fft = new FFT(this.img);
    fft.inverse();
    return fft.getImage();
}
```

And in general, isn't it a builder? So why are you returning Image and not this?

shimoncohen reviewed (changes requested):
Why do you have multiple functions for the same functionality?
In Flips class, there are functions similar to this one:
https://github.com/nspitzern/Image-Process-Project/blob/69a4fa27c4815476659e5632348ed30d180cc41f/src/ImageProcess/Flips.java#L25
Why not make a general function? Something like the following:
```java
static Image flip(Image img, Dimension d, Degrees degrees, Channel c) {
    ...
}
```
Where:
Dimensions -> enum {X, Y}
Degrees-> int [0, 360]
Channel -> enum {RED, GREEN, BLUE}

nspitzern committed:
Changed Filter interface to IFilter.
Added continue application after closing JFrame of IMAGE.show() .
Added inverseFFT when calling getImage() from FFT.
